gammapy/data/tests/test_filters.py

```diff
@@ -48,12 +49,12 @@ def test_filter_events(observation):
 
     time_filter = Time([53090.12, 53090.13], format="mjd", scale="tt")
 
-    obs_filter = ObservationFilter(
-        event_filters=[custom_filter, region_filter], time_filter=time_filter
-    )
-
-    events = observation.events
-    filtered_events = obs_filter.filter_events(events)
+    with pytest.warns(GammapyDeprecationWarning):
+        obs_filter = ObservationFilter(
+            event_filters=[custom_filter, region_filter], time_filter=time_filter
+        )
+        events = observation.events
+        filtered_events = obs_filter.filter_events(events)
```

Comment:
The specific line that creates the deprecation warning is the `filtered_events`:

```suggestion
    obs_filter = ObservationFilter(
        event_filters=[custom_filter, region_filter], time_filter=time_filter
    )
    events = observation.events
    with pytest.warns(GammapyDeprecationWarning):
        filtered_events = obs_filter.filter_events(events)
```